cwms/catalog/catalog.py

```diff
@@ -0,0 +1,56 @@
+from typing import Optional, Dict
+
+import cwms.api as api
+from cwms.types import Data
+
+
+def get_catalog(dataset: str, params: Dict[str, Optional[str]]) -> Data:
+    """Retrieves filters to display a catalog of data.
+
+        Parameters
+            ----------
+            dataset: string
+                The type of data in the list. Valid values for this field
+                are:
+                    1. TIMESERIES
+                    2. LOCATIONS
+            params: dict
+                A dictionary containing the following keys:
+                    page: string
+                        The endpoint used to identify where the request is located.
+                    page_size: integer
+                        The entries per page returned. The default value is 5000.
+                    unit_system: string
+                        The unit system desired in response. Valid values for this
+                        field are:
+                            1. SI
+                            2. EN
+                    office: string
+                        The owning office of the timeseries group.
+                    like: string
+                        The regex for matching against the id
+                    timeseries_category_like: string
+                        The regex for matching against the category id
+                    timeseries_group_like: string
+                        The regex for matching against the timeseries group id
+                    location_category_like: string
+                        The regex for matching against the location category id
+                    location_group_like: string
+                        The regex for matching against the location group id
+                    bounding_office_like: string
+                        The regex for matching against the location bounding office
+
+            Returns
+            -------
+            response : dict
+                The JSON response containing the time series catalog.
+        """
+
+    if dataset is None:
+        raise ValueError("Cannot retrieve a time series for a catalog without a dataset")
+
+    endpoint = f"catalog/{dataset}"
+
+    # creates the dataframe from the timeseries data
+    response = api.get(endpoint=endpoint, params=params, api_version=1)
+    return Data(response, selector="time-series-catalog")
```

Comment:
is the selector of 'time-series-catalog' the same for the location catalog?  Please make sure to test both.  now that I see everything I think we remove the get_catalog function.  Move the get_locations_catalog and get_timeseries_catalog into the catalog.py and have them both just call the api.get...  That way you can have the selector correct for each individual function.  If you have questions on that let me know.

Reply:
I apologize for my oversight and understand the problem. An alternative solution I have thought of is creating a check that sees whether it is a timeseries, or locations call and changes the selector while keeping the rest of the code. As an example: 

# Code
```
selector = "catalog"
    if dataset == "TIMESERIES":
        selector = "time-series-" + selector
    elif dataset == "LOCATIONS":
        selector = "locations-"

    return Data(response, selector=selector)
```

This keeps each respective function in their respective files (get_locations_catalog being in the locations directory). I have no opinion whether they should be all in the catalog directory or in their respective so I can adjust to what you believe is best.

Reply:
lets put the functions in the catalog and leave them as stand along. Adding the if statement would work but adds complexity for testing.  Keeping the functions as stand alone I think is best.